### ISSUE TITLE
Include item change history in preview and DITA published outputs

### ISSUE THREAD

**Issue by pjmonks** [MauroDataMapper-NHSD/mdm-plugin-nhs-data-dictionary#122]
# Description

Merging item changes in Mauro will add "MERGE" references to the item history, e.g.:

![image](https://github.com/user-attachments/assets/a14755e6-5224-42f7-bc6c-6fc254d850e4)

These item change references need to be included in the Data Dictionary item page output, both in the HTML preview and DITA published output.

Below is a mock-up of what the item history could look like from Steve Bowring (NHSE):

![image](https://github.com/user-attachments/assets/a7ad4ba7-6fcf-4196-9a02-d8d88e28807a)

# Change Log

Each item page must now include a "jump link" in the right-most column to the "Change Log" section.

The Change Log section will be another expandable/collapsable will display:

1. A table (styled the same as other tables on the page) listing:
    1. The Change Request number
    2. The Change Request description (name)
    3. The Implementation Date (the "Effective Date" from the "Work Item" profile)
2. A paragraph line below the table with a hyperlink to an archive page. This links will be provided by NHSE, pointing to a website they maintain.

Each CR number will be a hyperlink to a URL maintained by NHSE, where the CR number acts as the reference to the correct page to redirect to.

# Item Pages

The following item page types will be updated, both for preview and DITA:

- Data Sets
- Data Elements
- Attributes
- Classes
- NHS Business Definitions
- Supporting Information
- XML Schema Constraints

For Data Sets, Mauro will be tracking item changes from the Data Set itself plus the specification components (tables, sub-groups, rows of Data Elements, etc). For example:

- Data Set - changed in CR0001
    - Table 1 - changed in CR0002
        - Row 1 - changed in CR0001
    - Table 2 - changed in CR0003
        - Row 1 - changed in CR0003
        - Row 2 - changed in CR0004

The change log for this data set would list:

- CR0001
- CR0002
- CR0003
- CR0004

In these cases, the change log table will be a combination of all CR references referenced within the Data Set.

# Related

- NHSDigital/DataDictionaryPublication#312

**Comment by stevebowring** [comment]
@pjmonks Peter - can you explain the very last sentence:  " In these cases, the change log table will be a combination of all CR references referenced within the Data Set."

**Comment by pjmonks** [comment]
@stevebowring I have updated the description with an example. What it is trying to explain is that Data Sets recorded in Mauro are actually a lot of Mauro item components built up to define one large page. Each item component has it's own history, so we need to display the entire change log for the whole Data Set.

For instance, if a Data Set was created in CR0001, then in CR0002 one single row was updated with a change (e.g. change the mandation from "Mandatory" to "Optional"), the change history is recorded on that one row. The page describing the Data Set covers everything about the data set though, so I think the best way is to combine all the distinct change request numbers together into one table.
